### PR TITLE
chore: Handle non-error rejections in sentry

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description
When a non-error rejection is sent to sentry, we extract the actual error message and log it to sentry.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/9740

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/utils/AppsmithUtils.tsx [hunk 1]
```diff
@@ -36,6 +36,23 @@ export const initializeAnalyticsAndTrackers = () => {
               return null;
             }
           }
+          // Handle Non-Error rejections
+          if (
+            event.exception &&
+            Array.isArray(event.exception.values) &&
+            event.exception.values[0].value &&
```
Comment (hetunandu) [inline app/client/src/utils/AppsmithUtils.tsx:43]: these lines are repeated from above. It is better if we make it DRY

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/utils/AppsmithUtils.tsx (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/utils/AppsmithUtils.tsx]: Done

**Review by arunvjn — APPROVED** [review]

**Pull request merged**